File: 1415. The k-th Lexicographical String of All Happy Strings of Length n/ThekthLexicographicalStringofAllHappyStringsofLengthn.java
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class ThekthLexicographicalStringofAllHappyStringsofLengthn {
    int n,k,l=0;
    String[] sol;
    Map<String, List<String>> map = new HashMap<>();
    
    void happyString(String s, String c){
        if (s.length() == n-1) {
            sol[l] = s+c;
            l++;
        }else{
            happyString(s+c, map.get(c).get(0));
            happyString(s+c, map.get(c).get(1));
        }
    }

    public String getHappyString(int n, int k) {
        this.n = n;
        map.put("a", Arrays.asList("b", "c"));
        map.put("b", Arrays.asList("a", "c"));
        map.put("c", Arrays.asList("a", "b"));
        int temp = 3*(int) (Math.pow(2, n-1));
        this.sol = new String[temp];
        if (sol.length<k){
            return "";
        }
        happyString("", "a");
        happyString("", "b");
        happyString("", "c");
        return sol[k-1];
    }
    public static void main(String[] args) {
        ThekthLexicographicalStringofAllHappyStringsofLengthn tLengthn = new ThekthLexicographicalStringofAllHappyStringsofLengthn();
        System.out.println(tLengthn.getHappyString(3, 9));
    }
}

File: 1415. The k-th Lexicographical String of All Happy Strings of Length n/The k-th Lexicographical String of All Happy Strings of Length n.py
n = 6
k = 9
d = {'a':['b','c'], 'b':['a','c'], 'c':['a','b']}
solK = 3*(2**(n-1))
sol = []
def happyString(s,c):
    if len(s) == n-1:
        sol.append(s+c)
    else:
        happyString(s+c,d[c][0])
        happyString(s+c,d[c][1])
happyString("",'a')
happyString("",'b')
happyString("",'c')
print(sol[93],solK)